ir/type.cpp

```diff
@@ -622,6 +630,30 @@ StateValue AggregateType::extract(const StateValue &val,
                             (val.non_poison << idx).extract(h, l) });
 }
 
+StateValue AggregateType::update(const StateValue &val,
+                                 const StateValue &n,
+                                 const expr &index) const {
+  auto &elementTy = *children[0];
+  for (unsigned i = 1; i < elements; ++i) {
+    assert(elementTy.bits() == children[i]->bits());
+  }
+
+  unsigned bw_elem = elementTy.bits();
+  unsigned bw_val = val.value.bits();
```

Comment:
just bits() (which is equal to val.bits() and equal to val.value.bits()).  Assert below isn't needed

Reply:
fixed:)